build.gradle

```diff
@@ -60,3 +61,27 @@ jacocoTestReport {
 }
 
 build.dependsOn jacocoTestReport
+
+pmd {
+    toolVersion = '5.4.1'
+    ignoreFailures = false
+    ruleSetFiles = files('./code_quality_tools/pmd.xml')
+}
+
+task('pmd', type: Pmd, dependsOn: 'assemble') {
+    description = 'Run pmd'
+    group = 'verification'
+
+    ruleSets = []
+
+    source = fileTree('src/main/java')
+    include '**/*.java'
+    exclude '**/gen/**'
+
+    reports {
+        html.enabled = true
+        xml.enabled = true
+    }
+}
+
+build.dependsOn 'pmd'
```

Comment:
Yes, please remove this for now. We'll run PMD in our IDEs and post fixes for it. Once the master builds without errors, this can be re-enabled.